config/application.rb

```diff
@@ -87,9 +87,18 @@ class Application < Rails::Application
 
     config.action_controller.action_on_unpermitted_parameters = :raise
 
-    config.after_initialize do
-      # Token used to request badges
-      unless ENV['PRECOMPILE']
+    if !Rails.env.test? && !ENV['PRECOMPILE'] && ENV['SERVER_MODE'] && Job.table_exists? && Deploy.table_exists?
+      initializer :execute_job, after: :set_routes_reloader_hook do # flowdock uses routes: run after the routes are loaded
+        JobExecution.enabled = true
+        JobExecution.stop_running_jobs
+        JobExecution.start_pending_deploys
+        RestartSignalHandler.listen
```

Comment:
afaik this should only happen in server mode too ... so moving this into the block